src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { forwardRef, useContext } from 'react';
+import PropTypes from 'prop-types';
+import usePortal from 'react-cool-portal';
+import styled from 'styled-components';
+import { isNotUndefined } from 'ramda-adjunct';
+
+import { ModalProps } from './Modal.types';
+import { ModalWidthSizes } from './Modal.enums';
+import { useLockBodyScroll } from '../../hooks/useLockBodyScroll';
+import { useOuterClick } from '../../hooks/useOuterCallback';
+import { Icon } from '../Icon';
+import { Padbox } from '../layout';
+import { H3 } from '../typography';
+import {
+  createPadding,
+  getBorderRadius,
+  getColor,
+  getDepth,
+  getFontSize,
+  pxToRem,
+} from '../../utils';
+import { mergeRefs } from '../../utils/mergeRefs';
+import { SpaceSizes } from '../../theme';
+import { DSContext } from '../../theme/DSProvider/DSProvider';
+import { SSCIconNames } from '../../theme/icons/icons.enums';
+
+const widthVariants = {
+  [ModalWidthSizes.xs]: 320,
+  [ModalWidthSizes.sm]: 430,
+  [ModalWidthSizes.md]: 600,
+  [ModalWidthSizes.lg]: 900,
+};
+
+const Overlay = styled.div`
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  top: 0;
+  left: 0;
+  background: rgba(0, 0, 0, 0.6);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  z-index: ${getDepth('modal')};
+`;
+
+const BaseModal = styled.div<{ $width: number }>`
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  max-height: 66vh;
+  width: ${({ $width }) => pxToRem($width)};
```

Comment:
I missed that spot. We need to have modal horizontally resizable so we need to set width to 100% and given sizes as maximal width
```suggestion
  width: 100%;
  max-width: ${({ $width }) => pxToRem($width)};
```

Reply:
Good catch. Fixed 🙂